### PR TITLE
API diff between .NET 9 Preview 6 and .NET 9 Preview 7

### PR DESCRIPTION
**Pull request by carlossanlop**
Repo area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @lonitra @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @artl93 @jeffschwMSFT @jeffhandley @karelz @ericstj @stephentoub @SamMonoRT 

Libraries area owners:
- System @dotnet/area-system-runtime 
- System.Data @SamMonoRT @ajcvickers @DavoudEshtehari @David-Engel @roji
- System.Diagnostics @dotnet/area-system-diagnostics @dotnet/area-system-diagnostics-metric 
- System.Formats.Tar @dotnet/area-system-formats-tar 
- System.Net @dotnet/ncl 
- System.Numerics @dotnet/area-system-numerics 
- System.Runtime.InteropServices @dotnet/interop-contrib 
- System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics 
- System.Security @dotnet/area-system-security 
- System.Text.Json @dotnet/area-system-text-json 
- System.Threading @mangod9 @kouvel 

**Known api-diff tool issues**:

If yo usee any of these, please provide a GitHub suggestion in this PR to correct it: 

- The AsmDiff tool sometimes skips adding `{` or `{}` [arcade#10981](https://github.com/dotnet/arcade/issues/10981)(https://github.com/dotnet/arcade/issues/13814) 
- RequiresPreviewAttribute not getting auto added [arcade#14498](https://github.com/dotnet/arcade/issues/14498)
- Changes to `protected internal` visibility should stop showing. https://github.com/dotnet/arcade/issues/14579
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect: [arcade#13814]

### CODE REVIEW

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Formats.Tar.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# System.Formats.Tar
+
+``` diff
+ namespace System.Formats.Tar {
+     public abstract class TarEntry {
++        public long DataOffset { get; }
```
Comment (carlossanlop) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Formats.Tar.md:6]: System.Formats.Tar LGTM but I cannot sign-off my own PR.

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Numerics.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+# System.Numerics
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Numerics.md:1]: System.Numerics LGTM

CC. @dotnet/area-system-numerics

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Runtime.Intrinsics.Arm.md [hunk 3]
```diff
@@ -0,0 +1,477 @@
+# System.Runtime.Intrinsics.Arm
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Runtime.Intrinsics.Arm.md:1]: System.Runtime.Intrinsics.Arm LGTM

CC. @dotnet/area-system-runtime-intrinsics, @dotnet/arm64-contrib

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Runtime.Intrinsics.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# System.Runtime.Intrinsics
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Runtime.Intrinsics.md:1]: System.Runtime.Intrinsics LGTM

CC. @dotnet/area-system-runtime-intrinsics

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# System
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.md:1]: System LGTM

CC. @dotnet/area-system-runtime

Comment (tannergooding) [reply release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.md:1]: cc. @dotnet/area-system-datetime for the TimeSpan changes

**Review by tannergooding — APPROVED** [review]
My areas LGTM

**Review by eiriktsarpalis — APPROVED** [review]
STJ changes LGTM

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Data.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# System.Data
+
+``` diff
+ namespace System.Data {
+     public enum SqlDbType {
++        Json = 35,
```
[the remark this answers is not in the document]
Comment (roji) [reply release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Data.md:6]: Same

**Review by roji — APPROVED** [review]
OK for the System.Data changes

**Comment by tarekgh** [comment]
```
System.Diagnostics.Metrics
System.Environment
System.TimeSpan
```

LGTM

**Comment by kouvel** [comment]
System.Threading LGTM

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Security.Cryptography.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+-    [UnsupportedOSPlatformAttribute("ios")]
+-    [UnsupportedOSPlatformAttribute("tvos")]
+-    public sealed class AesGcm : IDisposable
++    [SupportedOSPlatformAttribute("ios13.0")]
++    [SupportedOSPlatformAttribute("maccatalyst")]
++    [SupportedOSPlatformAttribute("tvos13.0")]
++    [UnsupportedOSPlatformAttribute("browser")]
++    [UnsupportedOSPlatformAttribute("ios")]
++    [UnsupportedOSPlatformAttribute("tvos")]
++    public sealed class AesGcm : IDisposable
```
Comment (bartonjs) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Security.Cryptography.md:15]: The diff is weird... lines 5-7 and 12-14 are "remove this, and put it back", so could be eliminated.  Line 8/15 is just "context", not "diff".  But doesn't really matter.

**Review by bartonjs — APPROVED** [review]
System.Security LGTM, modulo some inefficient diffs.

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Net.Http.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+     public abstract class HttpContent : IDisposable {
++        public Task LoadIntoBufferAsync(long maxBufferSize, CancellationToken cancellationToken);
++        public Task LoadIntoBufferAsync(CancellationToken cancellationToken);
+     }
+     [UnsupportedOSPlatformAttribute("browser")]
+     public sealed class SocketsHttpHandler : HttpMessageHandler {
++        public bool EnableMultipleHttp3Connections { get; set; }
+     }
+ }
+```
```
Comment (rzikm) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Net.Http.md:14]: System.Net.* LGTM

File: release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Security.Cryptography.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+-    public sealed class ChaCha20Poly1305 : IDisposable
++    [SupportedOSPlatformAttribute("ios13.0")]
++    [SupportedOSPlatformAttribute("maccatalyst")]
++    [SupportedOSPlatformAttribute("tvos13.0")]
++    [UnsupportedOSPlatformAttribute("browser")]
++    [UnsupportedOSPlatformAttribute("ios")]
++    [UnsupportedOSPlatformAttribute("tvos")]
++    public sealed class ChaCha20Poly1305 : IDisposable
+     [FlagsAttribute]
+     public enum CngKeyCreationOptions {
```
Comment (carlossanlop) [inline release-notes/9.0/preview/preview7/api-diff/Microsoft.NETCore.App/9.0-preview7_System.Security.Cryptography.md:28]: This should fix it, @bartonjs 

```suggestion
 namespace System.Security.Cryptography {
+    [SupportedOSPlatformAttribute("ios13.0")]
+    [SupportedOSPlatformAttribute("maccatalyst")]
+    [SupportedOSPlatformAttribute("tvos13.0")]
     [UnsupportedOSPlatformAttribute("browser")]
     [UnsupportedOSPlatformAttribute("ios")]
     [UnsupportedOSPlatformAttribute("tvos")]
     public sealed class AesGcm : IDisposable
+    [SupportedOSPlatformAttribute("ios13.0")]
+    [SupportedOSPlatformAttribute("maccatalyst")]
+    [SupportedOSPlatformAttribute("tvos13.0")]
     [UnsupportedOSPlatformAttribute("browser")]
     [UnsupportedOSPlatformAttribute("ios")]
     [UnsupportedOSPlatformAttribute("tvos")]
     public sealed class ChaCha20Poly1305 : IDisposable
     [FlagsAttribute]
     public enum CngKeyCreationOptions {
```

**Pull request merged**